api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPatientDaoImpl.java

```diff
@@ -104,4 +115,52 @@ protected String paramToProp(String param) {
 		
 		return super.paramToProp(param);
 	}
+	
+	private void handleNames(List<PropParam<?>> params, Criteria criteria) {
+		AtomicReference<StringAndListParam> name = new AtomicReference<>();
```

Comment:
So we need `AtomicReference`s here?

Reply:
Not necessarily, I used it to avoid duplicate `alias`. That was the first thing I thought of but I think there's a better of handling this. 

The goal here is to call `handleNames(criteria, name, given, family)` once in a query to avoid duplicate alias. Alternative is to call it multiple times with different alias each the method is called. Didn't like the query generated. I chose the first option of storing passed in values temporarily.

Reply:
Doesn't it just make sense to avoid the using the lambdas altogether:

```java
private void handleNames(List<PropParam<?>> params, Criteria criteria) {
	StringAndListParam name = null;
	StringAndListParam given = null;
	StringAndListParam family = null;

	for (PropParam<?> param : params) {
		switch (param.getPropertyName()) {
			case FhirConstants.NAME_PROPERTY:
				name = (StringAndListParam) param.getParam();
				break;
			case FhirConstants.GIVEN_PROPERTY:
				given = (StringAndListParam) param.getParam();
				break;
			case FhirConstants.FAMILY_PROPERTY:
				family = (StringAndListParam) param.getParam();
				break;
		}
	}

	handleNames(criteria, name, given, family);
}
```

Reply:
That make sense ☺️